tests/test_androidmodule_ctypes_finder.py

```diff
@@ -0,0 +1,155 @@
+
+# Import android._ctypes_library_finder module which we need to do isolated.
+# (since android.__init__ will crash if imported not on android!)
```

Comment:
I think we usually patch top level module with `mock.patch.dict('sys.modules' ...`

Reply:
I actually asked in `#python` and nobody suggested that :joy: (since I never patched a module before), although is it really any different from what I'm doing? (I'm also just inserting an item into sys.modules)

Reply:
OK so we have two things here. 
1) For the android module crashing, I think it's more conventional to just:
```python
sys.modules['android._android'] = mock.MagicMock()
```
Then the line above should work without crashing:
```python
from pythonforandroid.recipes.android.src.android import _ctypes_library_finder
```

2) For Python2 `src` not being a package, I would just:
```sh
touch pythonforandroid/recipes/android/src/__init__.py
```
I think it worth it.

These two together would reduce your 26 lines of code to just one `sys.modules['android._android'] = mock.MagicMock()` I think it's totally worth it

Reply:
@AndreMiras right, great points! I find the direct import `pythonforandroid.recipes.android.src.android` with touch a bit ugly, so I did *almost* all of what you suggested but kept a bit of a `sys.path` hack but it's still way shorter than before. What do you think about how it looks now? Is that any good?

Reply:
Yes it looks way better to me. Regarding the touch I realise I wasn't clear enough. I do mean to track it in git. Yes I agree touching dynamically would otherwise be too hacky

Reply:
as a permanent addition? not a fan, could confuse people wondering why that is a module when it doesn't look like one. I think containing this to the test itself is nicer. but as always, open for you/someone else making the final call on this

Reply:
Let's leave it as you have for now then